lib/plugins/package/lib/zipDirectory.js

```diff
@@ -1,16 +1,55 @@
 'use strict';
 
+const childProcess = require('child_process');
 const archiver = require('archiver');
 const BbPromise = require('bluebird');
 const path = require('path');
 const fs = require('fs');
 const globby = require('globby');
+const _ = require('lodash');
+const fileExistsSync = require('../../../utils/fs/fileExistsSync');
 
 module.exports = {
   zipDirectory(exclude, include, zipFileName) {
```

Comment:
There is here a major issue if the serverless.yml file has multiple functions and each function is packaged individually. The problem is that this function is called for all serverless functions before anything is zipped, but each call to `zipDirectory` overrides the `include`, `exclude`, and `zipFileName` properties of `this`. This means that the last call to `zipDirectory` "wins" and all serverless functions end up being deployed with the code of the last data set by `zipDirectory`. This basically breaks the deployment if the serverless functions are packaged individually.